sys/net/network_layer/ng_sixlowpan/iphc/ng_sixlowpan_iphc.c

```diff
@@ -592,7 +606,8 @@ bool ng_sixlowpan_iphc_encode(ng_pktsnip_t *pkt)
 
             ctx = ng_sixlowpan_ctx_lookup_addr(&unicast_prefix);
 
-            if ((ctx != NULL) && (ctx->prefix_len == ipv6_hdr->dst.u8[3])) {
+            if ((ctx != NULL) && (ctx->flags_id & NG_SIXLOWPAN_CTX_FLAGS_COMP) &&
+                (ctx->prefix_len == ipv6_hdr->dst.u8[3])) {
```

Comment:
It's documented below: This is compression for the [unicast prefix based multicast address](https://tools.ietf.org/html/rfc3306#section-4). The forth (!) octet is the prefix length of the unicast prefix.